addons/survey/models/survey_survey.py

```diff
@@ -389,6 +390,17 @@ def _check_scoring_after_page_availability(self):
                 survey_names="\n- ".join(failing.mapped('title')))
             )
 
+    @api.constrains('user_id', 'restrict_user_ids')
+    def _check_survey_responsible_access(self):
+        """ When:
+                - a survey access is restricted to a list of users
+                - and there is a survey responsible,
+                - and this responsible is not survey manager (just survey officer),
+            check the responsible is part of the list."""
+        for survey in self:
+            if survey.restrict_user_ids and survey.user_id and not survey.user_id.has_group('survey.group_survey_manager') and survey.user_id not in survey.restrict_user_ids:
+                raise ValidationError(_('This survey access is restricted. Make sure its responsible still has access to it.'))
```

Comment:
Could be interesting to add details in error (Survey %(survey_name)s ... responsible %(user_name)s ...), that way if you update in batch you know which one is failing.